File: src2/Person.java
public abstract class Person {
    // Attributes
    private String firstName;
    private String lastName;
    private Password password;
    private String schedule;
    
    // Constructor
    
    public Person(String firstName, String lastName) {
        this.setFirstName(firstName);
        this.setLastName(lastName);
    }
    
    public Person(String firstName, String lastName, Password password) {
        this.setFirstName(firstName);
        this.setLastName(lastName);
    	this.setPassword(password);
    }
    
    public Person()  {
    	
    }
    
    public abstract void viewSchedule();
    
	public String getFirstName() {
		return firstName;
	}
	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}
	public String getLastName() {
		return lastName;
	}
	public void setLastName(String lastName) {
		this.lastName = lastName;
	}
	public Password getPassword() {
		return password;
	}
	public void setPassword(Password password) {
		this.password = password;
	}

	public String getSchedule() {
		return schedule;
	}

	public void setSchedule(String schedule) {
		this.schedule = schedule;
	}
}

File: Iteration3/Python/Person.py
class Person:


    def __init__(self, first_name=None, last_name=None, password=None):

        self.first_name = first_name
        self.last_name = last_name
        self.password = password
        self.schedule = []  # Default to an empty schedule

    def view_schedule(self):

        if not self.schedule:
            print(f"{self.first_name} {self.last_name} has no schedule.")
        else:
            print(f"\nSchedule for {self.first_name} {self.last_name}:")
            for item in self.schedule:
                print(f"- {item}")

    def get_first_name(self):

        return self.first_name

    def set_first_name(self, first_name):

        self.first_name = first_name

    def get_last_name(self):

        return self.last_name

    def set_last_name(self, last_name):

        self.last_name = last_name

    def get_password(self):

        return self.password

    def set_password(self, password):

        self.password = password

    def get_schedule(self):

        return self.schedule

    def set_schedule(self, schedule):

        self.schedule = schedule
